testing/tests/DevExpress.ui.widgets/treeViewParts/selection.js

```diff
@@ -372,3 +372,29 @@ QUnit.test("items should be selectable after the search", function(assert) {
 
     assert.equal(clickHandler.callCount, 1, "click works");
 });
+
+var HAS_SELECTED_CHILD_CLASS = "dx-treeview-node-has-selected-items";
+
+QUnit.test("parent has a special class if node has selected child", function(assert) {
```

Comment:
More tests required:

1. Recursive selection is enabled, select a middle-level child. Check if sub-children and parents have this class
2. Select all
3. Recursive selection is enabled/Disabled. Select/Deselect branch
4. Virtual mode/Create children. Select a leaf, load children, check the state
5. Select a parent of the Invisible/disabled children
6. Selection state is assigned on init. Select a middle-level child with the recursive selection. Check the state. Add this test to the server-side rendering (markup) tests